resolver/hosts_file_resolver.go

```diff
@@ -4,47 +4,52 @@ import (
 	"context"
 	"fmt"
 	"net"
-	"os"
-	"time"
 
 	"github.com/0xERR0R/blocky/config"
+	"github.com/0xERR0R/blocky/lists"
 	"github.com/0xERR0R/blocky/lists/parsers"
 	"github.com/0xERR0R/blocky/model"
 	"github.com/0xERR0R/blocky/util"
+	"github.com/ThinkChaos/parcour"
+	"github.com/ThinkChaos/parcour/jobgroup"
 	"github.com/miekg/dns"
 	"github.com/sirupsen/logrus"
 )
 
 const (
 	// reduce initial capacity so we don't waste memory if there are less entries than before
 	memReleaseFactor = 2
+
+	producersBuffCap = 1000
 )
 
+type HostsFileEntry = parsers.HostsFileEntry
+
 type HostsFileResolver struct {
 	configurable[*config.HostsFileConfig]
 	NextResolver
 	typed
 
-	hosts splitHostsFileData
+	hosts      splitHostsFileData
+	downloader lists.FileDownloader
 }
 
-type HostsFileEntry = parsers.HostsFileEntry
-
-func NewHostsFileResolver(cfg config.HostsFileConfig) *HostsFileResolver {
+func NewHostsFileResolver(cfg config.HostsFileConfig, bootstrap *Bootstrap) (*HostsFileResolver, error) {
 	r := HostsFileResolver{
 		configurable: withConfig(&cfg),
 		typed:        withType("hosts_file"),
-	}
 
-	if err := r.parseHostsFile(context.Background()); err != nil {
-		r.log().Errorf("disabling hosts file resolving due to error: %s", err)
+		downloader: lists.NewDownloader(cfg.Loading.Downloads, bootstrap.NewHTTPTransport()),
+	}
 
-		r.cfg.Filepath = "" // don't try parsing the file again
-	} else {
-		go r.periodicUpdate()
+	err := cfg.Loading.StartPeriodicRefresh(r.loadSources, func(err error) {
+		r.log().WithError(err).Errorf("could not load hosts files")
+	})
+	if err != nil {
+		return nil, err
 	}
```

Comment:
The hosts file resolver will now behave like the blocking resolver and have a `startStrategy`.  
This is different from before: we were never trying again to parse the file if it failed at initialization.

I think this makes more sense now that we support remote files and multiple sources. Do you agree?

Reply:
Yes, it makes definitively more sense. This should be documented properly to avoid confusions

Reply:
This would also make more sense in the changelog IMO, not resolving this yet since if that's not easy to do I can add it in the docs.